tests/runtime/iqm/test_iqm_runtime.py

```
@@ -0,0 +1,626 @@
+
+    devices = provider.get_devices()
+    assert len(devices) == 1
+    assert devices[0].id == "garnet"
+    assert devices[0].profile["dut_label"] == FakeIQMClient.static_architectures["garnet"].dut_label
+
+
+def test_iqm_provider_missing_device(fake_symbols):
+    """Test missing IQM device lookup."""
+    provider = IQMProvider(url="https://demo.iqm.fi")
+
+    with pytest.raises(ResourceNotFoundError, match="Device 'fake-device' not found."):
```

Ruff (0.15.7)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)